src/include/miopen/scalar_mul.hpp

```diff
@@ -0,0 +1,125 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2024 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#pragma once
+
+#include <miopen/common.hpp>
+#include <miopen/miopen.h>
+
+#include <cmath>
+#include <limits>
+#include <cassert>
+
+template <typename T>
+inline bool isCloseToZero(T value)
+{
+    return std::abs(value) <= std::numeric_limits<T>::epsilon();
+}
+
+template <typename T>
+inline bool isCloseToOne(float value)
+{
+    return std::abs(value - T(1)) <= std::numeric_limits<T>::epsilon();
+}
+
+namespace miopen {
+template <int default_val = 1>
```

Comment:
Don't like this choice. Let's call the class `Scalar` and take default value in the constructor. If we have `Scalar<0>` and `Scalar<1>`, now these are two types and you cannot assign one to the other without explicitly defining some kind of constructor or assignment operator or type conversion operator. So, not having a template parameter makes life easier.